Tools/clinic/clinic.py

```diff
@@ -2412,26 +2412,22 @@ def _module_and_class(
         this function is only ever used to find the parent of where
         a new class/module should go.
         """
-        parent: Clinic | Module | Class
-        child: Module | Class | None
-        module: Clinic | Module
+        parent: Clinic | Module | Class = self
+        module: Clinic | Module = self
         cls: Class | None = None
-        so_far: list[str] = []
 
-        parent = module = self
-
-        for field in fields:
-            so_far.append(field)
+        for idx, field in enumerate(fields):
             if not isinstance(parent, Class):
-                child = parent.modules.get(field)
-                if child:
-                    parent = module = child
+                if field in parent.modules:
+                    module = parent.modules[field]
+                    parent = module
                     continue
-            child = parent.classes.get(field)
-            if not child:
-                fullname = ".".join(so_far)
+            if field in parent.classes:
+                cls = parent.classes[field]
+                parent = cls
```

Comment:
Same here, I actually find it more readable to have them both on one line:

```suggestion
                parent = cls = parent.classes[field]
```